homeassistant/components/eq3btsmart/climate.py

```diff
@@ -0,0 +1,391 @@
+    async_get,
+    format_mac,
+)
+from homeassistant.helpers.entity import EntityPlatformState
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+from homeassistant.helpers.event import async_call_later
+
+from .const import (
+    DEVICE_MODEL,
+    DOMAIN,
+    EQ_TO_HA_HVAC,
+    HA_TO_EQ_HVAC,
+    MANUFACTURER,
+    CurrentTemperatureSelector,
+    Preset,
+    TargetTemperatureSelector,
+)
+from .eq3_entity import Eq3Entity
+from .models import Eq3Config, Eq3ConfigEntry
+
+_LOGGER = logging.getLogger(__name__)
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    config_entry: ConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Handle config entry setup."""
+
+    eq3_config_entry: Eq3ConfigEntry = hass.data[DOMAIN][config_entry.entry_id]
+
+    async_add_entities(
+        [Eq3Climate(eq3_config_entry.eq3_config, eq3_config_entry.thermostat)],
+        update_before_add=False,
+    )
+
+
+class Eq3Climate(Eq3Entity, ClimateEntity):
+    """Climate entity to represent a eQ-3 thermostat."""
+
+    def __init__(self, eq3_config: Eq3Config, thermostat: Thermostat) -> None:
+        """Initialize the climate entity."""
+
+        super().__init__(eq3_config, thermostat)
+
+        self._thermostat.register_update_callback(self._on_updated)
+        self._thermostat.register_connection_callback(self._on_connection_changed)
+        self._target_temperature: float | None = None
+        self._is_available = False
+        self._cancel_timer: Callable[[], None] | None = None
+        self._attr_has_entity_name = True
+        self._attr_name = None
+        self._attr_hvac_mode = None
+        self._attr_supported_features = (
+            ClimateEntityFeature.TARGET_TEMPERATURE
+            | ClimateEntityFeature.PRESET_MODE
+            | ClimateEntityFeature.TURN_OFF
+            | ClimateEntityFeature.TURN_ON
+        )
+        self._attr_temperature_unit = UnitOfTemperature.CELSIUS
+        self._attr_precision = PRECISION_TENTHS
+        self._attr_hvac_modes = list(HA_TO_EQ_HVAC.keys())
+        self._attr_min_temp = EQ3BT_OFF_TEMP
+        self._attr_max_temp = EQ3BT_MAX_TEMP
+        self._attr_preset_modes = list(Preset)
+        self._attr_unique_id = format_mac(self._eq3_config.mac_address)
+        self._attr_should_poll = False
+        self._was_connected: bool = False
+        self._firmware_version: str | None = None
+        self._attr_device_info: DeviceInfo = DeviceInfo(
+            name=self._eq3_config.name,
+            manufacturer=MANUFACTURER,
+            model=DEVICE_MODEL,
+            identifiers={(DOMAIN, self._eq3_config.mac_address)},
+            sw_version=self._firmware_version,
+            connections={(CONNECTION_BLUETOOTH, self._eq3_config.mac_address)},
+        )
+
+    async def async_added_to_hass(self) -> None:
+        """Run when entity about to be added to hass."""
+
+        asyncio.get_event_loop().create_task(self._async_scan_loop())
```

Comment:
Please use the Home Assistant api to create the task

`hass.async_create_bakcground_task` or `hass.async_create_task`

Creating a task without holding a strong reference to it may result in it being garbage collected before it finishes

https://docs.python.org/3/library/asyncio-task.html#asyncio.create_task

> Important Save a reference to the result of this function, to avoid a task disappearing mid-execution. The event loop only keeps weak references to tasks. A task that isn’t referenced elsewhere may get garbage collected at any time, even before it’s done. For reliable “fire-and-forget” background tasks, gather them in a collection:


This is not needed if you use the Home Assistant apis